app/controllers/posts_controller.rb

```diff
@@ -25,6 +25,15 @@ def create
     end
   end
 
+  def destroy
+    @post = Post.find(params[:id])
+    @post.destroy
+    respond_to do |format|
+      format.html { redirect_to user_posts_path, notice: 'Post successfully Deleted.' }
+      format.json { head :no_content }
+    end
+  end
+
```

Comment:
- [ ] kindly add post-authorization in your post controller i.e `load_and_authorize_resource`